### DISCUSSION TITLE
Extended Advertising and requestLEScan

### DISCUSSION THREAD

**Discussion by skygreenslade** [capacitor-community/bluetooth-le#602] [Q&A] [answered]
I am trying to use extended advertising to permit the broadcast of some data in the advertising packet that can be read by `requestLEScan`. Legacy advertising packets are not large enough to include the data, short name, and UUID of my device.

I cannot seem to pick up extended advertising packets in any capacity with `requestLEScan`. If I condense the name and data of my broadcast to use a legacy advertisement then it is picked up. 

Are extended advertisements supported? Is this a bug, or is there some way that I can get this to work?

**Comment by pwespi** [comment, marked as answer]
Hello, thank you for your interest in this plugin.

Extended advertisements are currently not supported by this plugin.

On Android, this would require to [set legacy to false](https://developer.android.com/reference/android/bluetooth/le/ScanSettings.Builder#setLegacy(boolean)) in the scan settings, which is currently not exposed by this plugin. The required API was added in API level 26, however Capacitor 5 currently has support down to API level 22. So this wouldn't work on all Android version currently supported by this plugin.

On iOS I'm not aware of any API for extended advertisements.

**Reply by skygreenslade** [reply]
Thanks for the reply! I appreciate the thorough explanation.

**Reply by ahoyahoy** [reply]
@pwespi Hi :)
any changes here after the release of version 7?

**Reply by peitschie** [reply]
Hi @ahoyahoy 

No patches have been proposed to add this feature (although one would be welcome!)

**Comment by jadejr** [comment]
I don't know enough about how the differences between different android versions from different OEMs. Would that mean this code wouldn't work the same on any non-samsung device? Anybody with any legacy advertising devices wanna test it? It works on my devices on this tablet (samsung galaxy s6 lite)

```kotlin
            scanSettings.setScanMode(scanMode)
            // NEW
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                scanSettings.setLegacy(false)
            }
```

It could also be wrapped in config option like `disableLegacyAdvertising` or `allowExtendedAdvertising` or some such.

**Reply by jadejr** [reply]
i misunderstood `setLegacy` the first time around. I think either `disableLegacyScanner` or  `enableLegacyScanner` are the ways to go.

Seeing issues like this https://github.com/PhilipsHue/flutter_reactive_ble/issues/654 suggests that it should stay conditional in some manner.
